lib/import_js/emacs_editor.rb

```diff
@@ -0,0 +1,145 @@
+# encoding: utf-8
+module ImportJS
+end
+
+class ImportJS::EmacsEditor
+  attr_accessor :current_word
+
+  def initialize
+    loop do
+      input = gets.chomp
+      command, value, path = input.split(':')
+
+      begin
+        @path = path
+        @file = File.readlines(path).map(&:chomp)
+
+        case command
+        when 'import'
+          @current_word = value
+          ImportJS::Importer.new(self).import
+          write_file
+          puts 'import:success'
+        when 'goto'
+          ImportJS::Importer.new(self).goto
+        end
+      rescue Exception => e
+        puts e.inspect
+      end
+    end
+  end
+
+  def write_file
+    new_file = File.open(@path, 'w')
+    @file.each {|line| new_file.puts(line) }
+    new_file.close
+  end
+
+  # Open a file specified by a path.
+  #
+  # @param file_path [String]
+  def open_file(file_path)
+    puts file_path
+  end
+
+  # Display a message to the user.
+  #
+  # @param str [String]
+  def message(str)
+    puts str
+  end
+
+  # Read the entire file into a string.
+  #
+  # @return [String]
+  def current_file_content
+    @file.join('\n')
```

Comment:
I forget where this is used, but don't you need to surround this with double quotes? `"\n"`